WikipediaLibrary/TWLight
# Kgraessle/t357057

[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
RTL user experience bugs for The Wikipedia Library

- Fixed twlight_static.sh to generate RTL css by moving the command to run it to the twlight_docker_entrypoint.sh script instead of in the Dockerfile. 
- Added generated files to the .gitignore. 
- Fixed RTL css issues with text getting cut off or overflowing parent container.

Bug: T357057

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
Fixes The Wikipedia Library RTL bugs reported in https://phabricator.wikimedia.org/T357057. 

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
https://phabricator.wikimedia.org/T357057

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
Since it was mostly fixing styling issues I did not add tests. 

[//]: # (- Can this change be tested manually? How?)
Yes- start the application with Docker and switch your language settings to a language that reads right to left. 
It's also probably worth manually testing non right to left settings as well. 
Steps to Reproduce: 

1. Start the app and login. 
2. Change your language in the Profile language dropdown to a RTL language (Hebrew and Arabic were the languages I used to test). 
3. Logout. 
4. View home page and ensure the text is not being cut off on the landing page.
5. Log back in to your homepage and resize the window. The two filter headers should not collide in the filter nav. 
6. Go to profile settings page and resize, ensure the two cards with the file icon and directory icon are not overflowing their containers. 
7. Go to the suggestion page, ensure the upvote button is not overflowing it's container. 

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)

## Issue 1
### Before
The RTL text was being cutoff like so: 
![image](https://github.com/WikipediaLibrary/TWLight/assets/30132257/8dc1c20f-d62b-4702-affd-0a89969f851b)

### After
Now, it looks like the following: 
![Screenshot 2024-05-10 at 11 25 13 AM](https://github.com/WikipediaLibrary/TWLight/assets/30132257/0856033c-0a48-4e59-ae74-bf4f9a0af6f2)

## Issue 2
### Before
The second issue I fixed is in the filter panel the 'Filters' and 'Reset Filters' text colliding. 
![image (1)](https://github.com/WikipediaLibrary/TWLight/assets/30132257/8f8bf7aa-8b4b-4f5e-9820-0ad7f6f14fa4)

### After 
The text wraps to a new line instead of colliding. 
![Screenshot 2024-05-10 at 11 30 04 AM](https://github.com/WikipediaLibrary/TWLight/assets/30132257/0785753c-08c1-4e7d-a2a3-ca31f0066949)

## Issue 3
### Before
The text was overflowing it's parent container 
![image (2)](https://github.com/WikipediaLibrary/TWLight/assets/30132257/dd30d781-dd1e-4914-b8c8-15b2d793b8db)

### After 
The text wraps to a new line instead of colliding. 
![Screenshot 2024-05-10 at 11 32 21 AM](https://github.com/WikipediaLibrary/TWLight/assets/30132257/c4fb0738-e8bf-42bf-b78b-61848304b44e)

## Issue 4
### Before
The button was overflowing it's parent container. 
![image (3)](https://github.com/WikipediaLibrary/TWLight/assets/30132257/1cc18925-12b3-4d17-b4ac-c5e8bb6099a8)

### After 
![Screenshot 2024-05-10 at 11 34 51 AM](https://github.com/WikipediaLibrary/TWLight/assets/30132257/4f4aec1d-3c17-43f4-9407-ae611e8a22d1)


## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
